### PR TITLE
Make infoview always enabled.

### LINKED ISSUE #101: When disabled `infoview` module, throws `:lua` error after attempting to open
First noticed this because of the fact that [suggested mappings are set regardless of whether their corresponding modules are `enable = true`'d](https://github.com/Julian/lean.nvim/blob/afdd047db65da46780825e4b303b3dbabbebf6bf/lua/lean/init.lua#L15).

For example, if you set `enable = false` for the `infoview` module, if you then try `<LocalLeader>i` in a Lean file you'll get a `:lua` error.

~~I believe we should fix this on a fundamental level by disabling calls to the public functions of a module if it has been disabled - using metatables, I think this should be doable without having to change the functions themselves.~~
EDIT: see discussion below

### PR DESCRIPTION
**Pull request by rish987**
Closes #101.

### CODE REVIEW

**Comment by Julian** [comment]
Awesome thanks!

Maybe we should add the test for the bug? I.e. where autoopen = false, making sure opening the infoview works without errors?

File: lua/tests/infoview/run_spec.lua [hunk 1]
```diff
@@ -189,73 +189,16 @@ describe('infoview', function()
...
-    it('no auto-open for irrelevant file',
-    function(_)
-      local num_wins = get_num_wins()
-      vim.api.nvim_command("tabedit temp")
-      assert.is.equal(num_wins + 1, get_num_wins())
-      assert.is.equal(1, #vim.api.nvim_tabpage_list_wins(0))
-    end)
   end)
 
+  -- TODO
```
Comment (Julian) [inline lua/tests/infoview/run_spec.lua:194]: Is this still a TODO?

**Review by Julian — APPROVED** [review]
Looks great, thanks!

**Comment by rish987** [comment]
Thanks -- I'm actually still working on a more general refactor of the tests (and also regarding that TODO will put that in its own spec). Will update soon!

**Comment by rish987** [comment]
(actually, will make a separate PR/issue for this)

**Pull request merged**